plugins/gatsby-theme-iterative-docs/src/utils/front/Slugger.ts

```diff
@@ -0,0 +1,43 @@
+class Slugger {
+  separator: string
+  lowercase: boolean
+  slugs: Array<string>
+
+  constructor(options?: { separator?: string; lowercase?: boolean }) {
+    this.separator = options?.separator || '-'
+    this.lowercase = options?.lowercase === false ? false : true
+    this.slugs = []
+  }
+
+  slug(str: string) {
+    str = typeof str === 'string' ? str : ''
+    let slug = this.slugify(str)
+
+    if (this.lowercase) {
+      slug = slug.toLowerCase()
+    }
+    if (this.slugs.includes(slug)) {
+      throw new Error(`Duplicate slug: ${slug}`)
+    }
+    this.slugs.push(slug)
+    return slug
+  }
+  slugify = (str: string) => {
```

Comment:
Is there a reason why the `slugify` function is written in a different way in comparison with the other functions?

Reply:
Nope, Thanks for spotting will make it consistent.